packages/client/src/features/tools/marquee-mouse-tool.ts

```diff
@@ -0,0 +1,208 @@
+/********************************************************************************
+ * Copyright (c) 2021 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { inject, injectable } from "inversify";
+import { Action, BoundsAware, EnableDefaultToolsAction, isSelectable, KeyListener, KeyTool, Point, SEdge, SelectAction, SModelElement, SNode, TYPES } from "sprotty";
+import { DragAwareMouseListener } from "../../base/drag-aware-mouse-listener";
+import { GLSP_TYPES } from "../../base/types";
+import { IMouseTool } from "../mouse-tool/mouse-tool";
+import { CursorCSS, cursorFeedbackAction } from "../tool-feedback/css-feedback";
+import { IFeedbackActionDispatcher } from "../tool-feedback/feedback-action-dispatcher";
+import { DrawMarqueeAction, RemoveMarqueeAction } from "../tool-feedback/selection-tool-feedback";
+import { BaseGLSPTool } from "./base-glsp-tool";
+import { getAbsolutePosition, toAbsoluteBounds } from "../../utils/viewpoint-util";
+import { DOMHelper } from "sprotty/lib/base/views/dom-helper";
+
+@injectable()
+export class MarqueeMouseTool extends BaseGLSPTool {
+    static ID = "glsp.marquee-mouse-tool";
+
+    @inject(GLSP_TYPES.MouseTool) protected mouseTool: IMouseTool;
```

Comment:
The mouseTool,keyTool and feedbackActionDispatcher are already injected in the super class ('BaseGLSPTool'). No need to inject them here again